src/common/tab/tabView.component.tsx

```diff
@@ -7,64 +7,93 @@ import { useSearchParams } from 'react-router-dom';
 import { a11yProps, StyledTab } from '../../common/tab/tab.utils';
 import TabPanel from '../../common/tab/tabPanel.component';
 
+type AdditionalTabValues = 'Gallery' | 'Attachments';
+
 export interface TabViewProps<T> {
-  defaultTab: T;
+  defaultTab: T | AdditionalTabValues;
   ariaLabelPrefix: string;
   tabData: {
-    value: T;
+    value: T | AdditionalTabValues;
     icon?: React.ReactElement;
     component: React.ReactNode;
+    order?: number;
   }[];
   galleryEntityId?: string;
+  galleryOrder?: number;
   attachmentsEntityId?: string;
+  attachmentsOrder?: number;
 }
 
 function TabView<T>(props: TabViewProps<T>) {
   const {
     defaultTab,
     ariaLabelPrefix,
-    tabData,
+    tabData: initialTabData,
     galleryEntityId,
+    galleryOrder,
     attachmentsEntityId,
+    attachmentsOrder,
   } = props;
+
   const [searchParams, setSearchParams] = useSearchParams();
 
-  // Retrieve the tab value from the URL or default to the passed defaultTab prop
-  const urlTabValue = (searchParams.get('tab') as T) || defaultTab;
-  const [tabValue, setTabValue] = React.useState<T>(urlTabValue);
+  const urlTabValue =
+    (searchParams.get('tab') as T | AdditionalTabValues) || defaultTab;
+  const [tabValue, setTabValue] = React.useState<T | AdditionalTabValues>(
+    urlTabValue
+  );
+
+  React.useEffect(() => {
+    setTabValue(urlTabValue);
+  }, [urlTabValue]);
 
   const handleTabChange = React.useCallback(
     (_event: React.SyntheticEvent, newValue: T) => {
       setTabValue(newValue);
+
+      const updatedParams = new URLSearchParams(searchParams);
+
       if (newValue === defaultTab) {
-        searchParams.delete('tab');
-        setSearchParams(searchParams, { replace: true });
+        updatedParams.delete('tab');
       } else {
-        setSearchParams(
-          { tab: newValue as unknown as string },
-          { replace: true }
-        );
+        updatedParams.set('tab', newValue as unknown as string);
       }
+
+      setSearchParams(updatedParams);
     },
     [defaultTab, searchParams, setSearchParams]
   );
 
-  // Optionally add Gallery and Attachments tabs
-  if (galleryEntityId) {
-    tabData.push({
-      value: 'Gallery' as T,
-      icon: <CollectionsOutlinedIcon />,
-      component: <></>,
-    });
-  }
-
-  if (attachmentsEntityId) {
-    tabData.push({
-      value: 'Attachments' as T,
-      icon: <AttachmentOutlinedIcon />,
-      component: <></>,
-    });
-  }
+  const tabData = React.useMemo(() => {
```

Comment:
The plan is to use these tabs in other places aswell so that's why I made it optional (details panel and view tabs )

Reply:
Sure that's fine, I knew it would be used in the other places which is why I said it would be repeating, and would sacrifice that for not needing a memo, ordering or entity parameters for each.